src/mavsdk/plugins/camera/camera_impl.h

```diff
@@ -23,299 +28,369 @@ class CameraImpl : public PluginImplBase {
+    void set_setting_async(
+        int32_t camera_id, const Camera::Setting& setting, const Camera::ResultCallback& callback);
 
-    std::vector<Camera::SettingOptions> possible_setting_options();
+    std::pair<Camera::Result, Camera::Setting>
+    get_setting(int32_t camera_id, const Camera::Setting& setting);
 
     bool is_setting_range(const std::string& setting_id);
 
     Camera::CurrentSettingsHandle
     subscribe_current_settings(const Camera::CurrentSettingsCallback& callback);
     void unsubscribe_current_settings(Camera::CurrentSettingsHandle handle);
+
+    std::pair<Camera::Result, std::vector<Camera::Setting>> get_current_settings(int32_t camera_id);
+
     Camera::PossibleSettingOptionsHandle
     subscribe_possible_setting_options(const Camera::PossibleSettingOptionsCallback& callback);
     void unsubscribe_possible_setting_options(Camera::PossibleSettingOptionsHandle handle);
 
-    Camera::Result format_storage(int32_t storage_id);
-    void format_storage_async(int32_t storage_id, const Camera::ResultCallback callback);
+    std::pair<Camera::Result, std::vector<Camera::SettingOptions>>
+    get_possible_setting_options(int32_t camera_id);
+
+    Camera::Result format_storage(int32_t camera_id, int32_t storage_id);
+    void format_storage_async(
+        int32_t camera_id, int32_t storage_id, const Camera::ResultCallback& callback);
 
-    Camera::Result reset_settings();
-    void reset_settings_async(const Camera::ResultCallback callback);
+    Camera::Result reset_settings(int32_t camera_id);
+    void reset_settings_async(int32_t camera_id, const Camera::ResultCallback& callback);
 
     std::pair<Camera::Result, std::vector<Camera::CaptureInfo>>
-    list_photos(Camera::PhotosRange photos_range);
-    void
-    list_photos_async(Camera::PhotosRange photos_range, const Camera::ListPhotosCallback callback);
+    list_photos(int32_t camera_id, Camera::PhotosRange photos_range);
+    void list_photos_async(
+        int32_t camera_id,
+        Camera::PhotosRange photos_range,
+        const Camera::ListPhotosCallback& callback);
 
     CameraImpl(const CameraImpl&) = delete;
     CameraImpl& operator=(const CameraImpl&) = delete;
 
 private:
-    bool get_possible_setting_options(std::vector<std::string>& settings);
-    bool get_possible_options(const std::string& setting_id, std::vector<Camera::Option>& options);
+    // struct CaptureStatus {
+    //     // Duplicate right now.
+    //     Camera::Status data{};
+    //     std::unique_ptr<std::thread> fetch_thread{};
+
+    //    bool received_camera_capture_status{false};
+    //    bool received_storage_information{false};
+    //    int image_count{-1};
+    //    int image_count_at_connection{-1};
+    //    std::map<int, Camera::CaptureInfo> photo_list;
+    //    bool is_fetching_photos{false};
+    // };
+
+    // struct CaptureInfo {
+    //     int last_advertised_image_index{-1};
+    //     std::map<int, int> missing_image_retries{};
+    // };
```

Comment:
Is there another kind of way to tell the user that a photo has been taken or is it completely disabled?

Reply:
I want to add it back in, but I have nothing to test this with. I was hoping I could write the server part of it, and then test it against it once I have that.

It's not really something that is very common or standardized in MAVLink, right?

Reply:
I can't speak for "standardized" 🙈, but I think it's common for mission planning apps to expect some feedback from the camera, e.g. to know that photos are being taken more or less as expected. The updates are unreliable so some updates are missed, but the apps would generally rather have a reliable update than no update at all, I would say.

Reply:
Sure, I agree there.